moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -385,12 +404,16 @@ void ServoCalcs::calculateSingleIteration()
   }
 
   // If we should halt
-  if (!have_nonzero_command_)
+  if (!have_nonzero_command_ && !done_stopping_)
   {
-    suddenHalt(*joint_trajectory);
+    filteredHalt(*joint_trajectory);
     have_nonzero_twist_stamped_ = false;
     have_nonzero_joint_command_ = false;
   }
+  else
+  {
+    done_stopping_ = false;
```

Comment:
Wish there was a way to suggest changes to non-modified lines. Line 420 
```
  if (!have_nonzero_command_ && **done_stopping_** && (parameters_->num_outgoing_halt_msgs_to_publish != 0) &&
      (zero_velocity_count_ > parameters_->num_outgoing_halt_msgs_to_publish))
```

and line 442
```
  if (!have_nonzero_command_ && **done_stopping_**)
```

Need to check this var. Otherwise it can reach num_outgoing_halt_msgs_to_publish before filteredHalt actually completes and the last publishing point will be non-zero.